vent/controller/control_module.py

```diff
@@ -470,148 +335,219 @@ def __get_PID_error(self, ytarget, yis, dt):
         self._DATA_P = error_new
 
     def __calculate_control_signal_in(self):
+        """
+        Calculated the PID control signal with the error terms and the three gain parameters.
+        """
         self.__control_signal_in  = 0            # Some setting for the maximum flow.
         self.__control_signal_in +=  self.__KP*self._DATA_P
         self.__control_signal_in +=  self.__KI*self._DATA_I
         self.__control_signal_in +=  self.__KD*self._DATA_D
 
     def _get_control_signal_in(self):
-        ''' This is the PID controlled signal on the inspiratory side '''
+        ''' This is the controlled signal on the inspiratory side '''
         return self.__control_signal_in
 
     def _get_control_signal_out(self):
         ''' This is the control signal (open/close) on the expiratory side '''
         return self.__control_signal_out
 
-    def _PID_reset(self):
-        ''' resets the PID cycle to zero'''
+    def _control_reset(self):
+        ''' Resets the internal controller cycle to zero, i.e. this breath cycle re-starts.'''
         self.__cycle_start = time.time()
 
-    def _PID_update(self, dt):
+    def __test_for_alarms(self):
+        """
+        Implements tests that are to be executed in the main control loop:
+            - Test for HAPA
+            - Test for Technical Alert, making sure sensor values are plausible
+            - Test for Technical Alert, make sure continuous in contact
+        Currently: Alarms are time.time() of first occurance.
+        """
+        limit_hapa = 80                 # TODO: WHAT IS THE LIMIT?
+        limit_max_flows = 10            # If flows above that, hardware cannot be correct.
```

Comment:
These look like technical alarm limits to me -- they should be implemented in `alarm.__init__.py` as `AlarmType` s and conditions. otherwise you can't (and shouldnt be able to) create alarm objects

Reply:
I see. Thanks for implementing! I see the new classes, and `AlarmSeverity.TECHNICAL` everywhere